lib/iris/tests/__init__.py

```diff
@@ -205,6 +205,14 @@ def setUpClass(cls):
         iris.site_configuration['cf_profile'] = None
 
     def _assert_str_same(self, reference_str, test_str, reference_filename, type_comparison_name='Strings'):
+        # Work around to deal with alterations to attribute typing in
+        # netcdf4python, introduced in 1.1 and present in 1.2. see:
+        # https://github.com/Unidata/netcdf-c/issues/298
+        # https://github.com/Unidata/netcdf4-python/issues/529
+        # https://github.com/Unidata/netcdf4-python/issues/575
+        # amongst others for further details.
+        if type_comparison_name == 'CDL' and reference_str != test_str:
+            test_str = test_str.replace('\t\tstring ', '\t\t')
```

Comment:
@marqh I really don't like the fact that we're burying this patch in the bowels of the testing framework, where it will be forgotten and perhaps sting us some time (long?) in the future.

Is there an opportunity here to elevate this change to be a registered iris patch? A proper thing (a first class iris citizen) that isn't hidden, but something that is very visible, easy to control, govern and perhaps at some point retire or deprecate.

Specifically, I'm thinking of an iris patch module that contains all patch functionality that we currently apply in the code base. The application of such patching (to me) seems quite generic and specific to individual patching cases. But in this case I'm thinking of a specific patch decorator that we would apply to this specific method, where the patch decorator would do the regexp on the test_str, as required.

The specific  patch method could even only be applied for netCDF v1.1+ and contain all relevant documented details/history for other dev's in the future to understand the context of this specific patch ...

Thoughts? Or am I guilty of over thinking this ... I still believe it would be a simple change, but it's the patching strategy this is more interesting